src/Shared/AssemblyNameExtension.cs

```diff
@@ -186,7 +186,23 @@ internal static AssemblyNameExtension GetAssemblyNameEx(string path)
 #if !FEATURE_ASSEMBLYLOADCONTEXT
             try
             {
-                assemblyName = AssemblyName.GetAssemblyName(path);
+                try
```

Comment:
Is there any reason to make this a separate try statement? OutOfMemoryException isn't caught in the outer try statement, so if you catch it once, run the garbage collector, and try again, it'll have the same result.

Reply:
The retry can throw an exception that is caught by the outer handler.